src/engine/strat_engine/backstore/backstore.rs

```diff
@@ -215,96 +456,373 @@ impl Backstore {
+                        let pool_uuid = self.data_tier.block_mgr.pool_uuid();
+                        let bdm =
+                            BlockDevMgr::initialize(pool_uuid, paths, MIN_MDA_SECTORS, force)?;
+
+                        let linear = self.linear
+                            .take()
+                            .expect("cache_tier.is_none() <=> self.linear.is_some()");
+                        let (cache_tier, cache) = CacheTier::new(dm, bdm, linear)?;
+                        self.cache = Some(cache);
+
+                        let uuids = cache_tier
+                            .block_mgr
+                            .blockdevs()
+                            .iter()
+                            .map(|&(uuid, _)| uuid)
+                            .collect::<Vec<_>>();
+
+                        self.cache_tier = Some(cache_tier);
+
+                        Ok(uuids)
+                    }
+                }
+            }
+            BlockDevTier::Data => {
+                self.data_tier
+                    .add(dm, self.cache.as_mut(), self.linear.as_mut(), paths, force)
+            }
+        }
     }
 
-    /// Allocate space from the underlying device.
+    /// Allocate requested chunks from device.
+    /// Returns None if it is not possible to satisfy the request.
+    /// Each segment allocated is contiguous with its neighbors in the return
+    /// vector.
+    /// WARNING: All this must change when it becomes possible to return
+    /// sectors to the store.
     /// WARNING: metadata changing event
     pub fn alloc_space(&mut self, sizes: &[Sectors]) -> Option<Vec<Vec<(Sectors, Sectors)>>> {
-        self.data_tier.alloc_space(sizes)
+        if self.available() < sizes.iter().cloned().sum() {
+            return None;
+        }
+
+        let mut chunks = Vec::new();
+        for size in sizes {
+            chunks.push(vec![(self.next, *size)]);
+            self.next += *size;
+        }
+        Some(chunks)
     }
 
-    /// Return a reference to the blockdevs that form the basis of the
-    /// backstore.
+    /// Return a reference to all the devs that this pool has ownership of.
+    /// This includes blockdevs, cachedevs, any device about which information
+    /// may be placed on the D-Bus.
     pub fn blockdevs(&self) -> Vec<(DevUuid, &BlockDev)> {
-        self.data_tier.block_mgr.blockdevs()
+        match self.cache_tier {
+            Some(ref cache) => {
+                cache
+                    .blockdevs()
+                    .iter()
+                    .chain(self.data_tier.blockdevs().iter())
+                    .map(|&x| x)
+                    .collect()
+            }
+            None => self.data_tier.blockdevs(),
+        }
     }
 
-    /// The current capacity of all the blockdevs that make up the backstore.
-    pub fn current_capacity(&self) -> Sectors {
+    /// The current capacity of all the blockdevs in the data tier.
+    pub fn datadev_current_capacity(&self) -> Sectors {
```

Comment:
suggest rename to datatier_capacity, "tier" is more accurate than "dev" and "current" doesn't add much.

Reply:
I think it will make a difference when we resume worrying about the recorded size of the blockdevs, so I'ld rather keep it.

Reply:
datatier is fine, though.